test/test-functions

```diff
@@ -198,21 +300,28 @@ run_qemu() {
 
     if [[ "$LOOKS_LIKE_SUSE" ]]; then
         PARAMS+="rd.hostonly=0"
-    elif [[ "$LOOKS_LIKE_ARCH" ]]; then
-        PARAMS+="rw"
+    fi
+
+    local _end
+    if [[ ! "$INTERACTIVE_DEBUG" ]]; then
+        _end="systemd.wants=end.service"
     else
-        PARAMS+="ro"
+        _end=""
     fi
 
     KERNEL_APPEND="$PARAMS \
 root=/dev/sda1 \
+rw \
 raid=noautodetect \
 rd.luks=0 \
 loglevel=2 \
 init=$PATH_TO_INIT \
 console=$CONSOLE \
 selinux=0 \
 $_cgroup_args \
+SYSTEMD_UNIT_PATH=/usr/lib/systemd/tests/testdata/testsuite-$1.units:/usr/lib/systemd/tests/testdata/units: \
```

Comment:
For some strange reason this has no effect when running the integration tests on Arch Linux (or at least in the CentOS CI variant of it).

Logs from TEST-01-BASIC:

CentOS 7:
```
Kernel command line:  root=/dev/sda1 rw raid=noautodetect rd.luks=0 loglevel=2 init=/usr/lib/systemd/systemd console=ttyS0 selinux=0  SYSTEMD_UNIT_PATH=/usr/lib/systemd/tests/testdata/testsuite-01.units:/usr/lib/systemd/tests/testdata/units: systemd.unit=testsuite.target systemd.wants=testsuite-01.service systemd.wants=end.service debug systemd.log_level=debug systemd.log_target=console 
...
Looking for unit files in (higher priority first):
        /usr/lib/systemd/tests/testdata/testsuite-01.units
        /usr/lib/systemd/tests/testdata/units
        /etc/systemd/system.control
        /run/systemd/system.control
        /run/systemd/transient
        /run/systemd/generator.early
        /etc/systemd/system
        /etc/systemd/system.attached
        /run/systemd/system
        /run/systemd/system.attached
        /run/systemd/generator
        /usr/local/lib/systemd/system
        /usr/lib/systemd/system
        /run/systemd/generator.late
```

Arch Linux (with sanitizers):
```
Kernel command line:  root=/dev/sda1 rw raid=noautodetect rd.luks=0 loglevel=2 init=/usr/lib/systemd/systemd-under-asan console=ttyS0 selinux=0  SYSTEMD_UNIT_PATH=/usr/lib/systemd/tests/testdata/testsuite-01.units:/usr/lib/systemd/tests/testdata/units: systemd.unit=testsuite.target systemd.wants=testsuite-01.service systemd.wants=end.service  
...
Looking for unit files in (higher priority first):
        /etc/systemd/system.control
        /run/systemd/system.control
        /run/systemd/transient
        /run/systemd/generator.early
        /etc/systemd/system
        /etc/systemd/system.attached
        /run/systemd/system
        /run/systemd/system.attached
        /run/systemd/generator
        /usr/local/lib/systemd/system
        /usr/lib/systemd/system
        /run/systemd/generator.late
```

Arch Linux:
```
Kernel command line:  root=/dev/sda1 rw raid=noautodetect rd.luks=0 loglevel=2 init=/usr/lib/systemd/systemd console=ttyS0 selinux=0  SYSTEMD_UNIT_PATH=/usr/lib/systemd/tests/testdata/testsuite-01.units:/usr/lib/systemd/test
s/testdata/units: systemd.unit=testsuite.target systemd.wants=testsuite-01.service systemd.wants=end.service debug systemd.log_level=debug systemd.log_target=console 
...
Looking for unit files in (higher priority first):
        /etc/systemd/system.control
        /run/systemd/system.control
        /run/systemd/transient
        /run/systemd/generator.early
        /etc/systemd/system
        /etc/systemd/system.attached
        /run/systemd/system
        /run/systemd/system.attached
        /run/systemd/generator
        /usr/local/lib/systemd/system
        /usr/lib/systemd/system
        /run/systemd/generator.late
```

Reply:
Ah, I see. After some poking around, the issue is in the default initrd image.

By default, Arch Linux generates the initrd image with the `udev` hook (among others) included:

```
# grep ^HOOKS /etc/mkinitcpio.conf 
HOOKS=(base udev autodetect modconf block filesystems keyboard fsck)
```

However, in this case, `/sbin/init` is a symlink to a `busybox` binary, which then executes a simple init wrapper, that, in the end, executes systemd. Apparently, in this case, the set env variables from the kernel command line got lost around the way.

After switching from the `udev` hook to the `systemd` one (which [includes](https://wiki.archlinux.org/index.php/Mkinitcpio#Common_hooks) the `udev` one as well), everything works as expected and the test passes. I'll go ahead and update the CI scripts accordingly.

Reply:
Hopefully fixed by https://github.com/systemd/systemd-centos-ci/pull/230